storage/innobase/include/sync0policy.h

```diff
@@ -128,6 +128,18 @@ class MutexDebug {
 			return(std::string(msg.str()));
 		}
 
+		/** Synchronized m_thread_id getter */
+		os_thread_id_t get_thread_id() const
+		{
+			return my_atomic_loadlong(&m_thread_id);
+		}
+
+		/** Synchronized m_thread_id setter */
+		void set_thread_id(os_thread_id_t thread_id)
+		{
+                        my_atomic_storelong(&m_thread_id, thread_id);
+		}
+
```

Comment:
I don't really like when atomic operations are hidden behind getters/setters: it is not immediately obvious that this is atomic access and what memory barrier is being issued. Makes it a lot harder to interpret this code from concurrent point of view.